sdk/common/shared/src/main/scala/org/typelevel/otel4s/sdk/autoconfigure/AutoConfigure.scala

```diff
@@ -89,7 +89,7 @@ object AutoConfigure {
     * @tparam A
     *   the type of the component
     */
-  abstract class WithHint[F[_]: MonadThrow, A](
+  abstract class WithHint[F[_]: MonadCancelThrow, A](
       hint: String,
       configKeys: Set[Config.Key[_]]
   ) extends AutoConfigure[F, A] {
```

Comment:
The compiler can no longer lift an instance of `ApplicativeError[Resource[F, *], Throwable]`. I assume it's caused by:
- https://github.com/typelevel/cats-effect/pull/4128
- https://github.com/typelevel/cats-effect/pull/4180